src/BloomExe/Browser.cs

```diff
@@ -419,21 +419,23 @@ private void PasteFilteredText(bool removeSingleLineBreaks)
 
 		void OnShowContextMenu(object sender, GeckoContextMenuEventArgs e)
 		{
+			MenuItem FFMenuItem = null;
 			Debug.Assert(!InvokeRequired);
 			if (ContextMenuProvider != null)
 			{
-				ContextMenuProvider(e);
-#if DEBUG
-				e.ContextMenu.MenuItems.Add("Open Page in Firefox (which must be in the PATH environment variable)", new EventHandler(OnOpenPageInSystemBrowser));
-#endif
-				if (ContextMenuProvider(e))
-					return; // only the provider's items
+				var replacesStdMenu = ContextMenuProvider(e);
+				FFMenuItem = e.ContextMenu.MenuItems.Add("Open Page in Firefox (which must be in the PATH environment variable)",
```

Comment:
If I understand the issue (and I may not) I think this should be below our radar at the moment.